SWATCH-3978: Fix component tests dependency issue for swatch-metrics-hbi

lindseyburnett opened the pull request:
Jira issue: SWATCH-3978

## Description

Fixes component tests pipeline failure for swatch-metrics-hbi by adding the main module and all its dependencies to the component-tests Maven profile to ensure proper build order.

The component tests import classes from the main swatch-metrics-hbi module (HbiHostCreateUpdateEvent, Host, etc.) and depend on the main module JAR. However, the component-tests profile wasn't building the main module or its dependencies first, causing multiple "Could not find artifact" errors.

**Required Dependencies Added:**
- swatch-metrics-hbi (main module)
- swatch-common-clock, swatch-common-export, swatch-common-config-workaround
- swatch-common-resteasy, swatch-common-resteasy-client, swatch-common-panache
- swatch-common-kafka, swatch-common-smallrye-fault-tolerance, swatch-common-splunk
- swatch-common-trace-response, swatch-common-security, swatch-common-health
- swatch-common-models, swatch-common-testcontainers
- swatch-model-events, swatch-product-configuration

**Architectural Concerns:** This approach significantly increases coupling between the test framework and service modules, making 16 additional modules available to all other component tests. This goes against the goal of better separation of concerns between services. I'm putting together some notes/thoughts and some team discussion points.

## Testing

### Setup
1. Checkout this branch
1. Invalidate m2 cache: `rm -rf ~/.m2/repository/com/redhat/swatch/`
1. Run component tests to verify they pass

### Steps
1. `./mvnw clean install -P component-tests -DskipTests`

### Verification
1. Build completes successfully without "Could not find artifact" errors
1. All swatch-metrics-hbi dependencies build before the main module
1. swatch-metrics-hbi module builds before its component tests

### Reproduce Original Issue
To reproduce the original issue before this fix:
1. `git checkout main`
1. `rm -rf ~/.m2/repository/com/redhat/swatch/swatch-metrics-hbi/`
1. `./mvnw clean install -P component-tests -DskipTests`
1. Should fail with "Could not find artifact com.redhat.swatch:swatch-metrics-hbi:jar:1.1.0-SNAPSHOT"

lindseyburnett committed:
SWATCH-3978: Fix component tests dependency issue for swatch-metrics-hbi

Add swatch-metrics-hbi main module to component-tests profile to ensure it builds before its component tests. This creates coupling between the test framework and the swatch-metrics-hbi module, which is not ideal for separation of concerns.

The component tests import classes from the main module (HbiHostCreateUpdateEvent, Host, etc.) and depend on the main module JAR, but the component-tests profile wasn't building the main module, causing: 'Could not find artifact com.redhat.swatch:swatch-metrics-hbi:jar:1.1.0-SNAPSHOT'

This approach has drawbacks:
- Creates coupling between test framework and service module
- Makes swatch-metrics-hbi classes available to other component tests
- Goes against goal of better separation of concerns between services

lindseyburnett committed:
Add all swatch-metrics-hbi dependencies to component-tests profile

The swatch-metrics-hbi module requires many common modules that weren't included in the component-tests profile, causing build failures in the pipeline. Added all required dependencies:

- swatch-common-clock
- swatch-common-export
- swatch-common-config-workaround
- swatch-common-resteasy
- swatch-common-resteasy-client
- swatch-common-panache
- swatch-common-kafka
- swatch-common-smallrye-fault-tolerance
- swatch-common-splunk
- swatch-common-trace-response
- swatch-common-security
- swatch-common-health
- swatch-common-models
- swatch-common-testcontainers
- swatch-model-events
- swatch-product-configuration

This significantly increases the coupling between the test framework and service modules, making these dependencies available to all component tests.

InsightsDroid commented:
### IQE Tests Summary Report
 301 passed, 2 skipped, 181 deselected, 70 warnings in 2188.40s (0:36:28) 
Result: Success
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions

lindseyburnett committed:
Create standalone swatch-model-metrics-hbi shared module

Replace massive component-tests dependency list with lightweight shared module approach.

**New Architecture:**
- Created swatch-model-metrics-hbi shared module containing only data models needed by component tests
- Moved OpenAPI generation for FlushResponse to shared module
- Extracted essential DTOs: HbiHostCreateUpdateEvent, HbiHost, HbiEvent, etc.
- Extracted simple fact models: SystemProfileFacts, RhsmFacts, Host, NormalizedEventType
- Added HbiEventConstants for shared constants

**Component Tests Profile Improvement:**
- Before: 16+ heavy modules (swatch-common-*, swatch-product-configuration, etc.)
- After: 3 lightweight modules (swatch-model-billable-usage, swatch-model-metrics-hbi, swatch-test-framework)

**Benefits:**
- ✅ Eliminates coupling between test framework and 16+ service modules
- ✅ Component tests are now self-contained with minimal dependencies
- ✅ Follows established swatch-model-* pattern like swatch-model-billable-usage
- ✅ Addresses architectural concerns about service separation
- ✅ Build time improvements through reduced dependency graph

**Architecture Decision:**
Extracted only simple data models and constants that component tests actually import. Left complex business logic (normalizers, services) in main module to avoid dependency sprawl.

lindseyburnett commented:
Closing this PR in favor of two new PRs that better organize the different approaches:

🔴 **PR #5125**: Shows the coupling problem with the band-aid approach (16+ modules)
🟢 **PR #5126**: Shows the clean architectural solution (standalone shared module)

This separation makes it easier to compare the approaches and understand why the architectural solution is preferred.
